sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerClient.cs

```diff
@@ -0,0 +1,254 @@
+        public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential, ConfidentialLedgerClientOptions options = null)
+        {
+            if (ledgerUri == null)
+            {
+                throw new ArgumentNullException(nameof(ledgerUri));
+            }
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+
+            options ??= new ConfidentialLedgerClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            clientDiagnostics = _clientDiagnostics;
+            _tokenCredential = credential;
+            var authPolicy = new BearerTokenAuthenticationPolicy(_tokenCredential, AuthorizationScopes);
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() },
+                new HttpPipelinePolicy[] { authPolicy },
+                new ResponseClassifier());
+            this.ledgerUri = ledgerUri;
+            apiVersion = options.Version;
+            DefaultPollingInterval = options.WaitForCompletionDefaultPollingInterval;
+        }
+
+        /// <summary> Posts a new entry to the ledger. A sub-ledger id may optionally be specified. </summary>
+        /// <remarks>
+        /// Schema for <c>Request Body</c>:
+        /// <list type="table">
+        ///   <listheader>
+        ///     <term>Name</term>
+        ///     <term>Type</term>
+        ///     <term>Required</term>
+        ///     <term>Description</term>
+        ///   </listheader>
+        ///   <item>
+        ///     <term>contents</term>
+        ///     <term>string</term>
+        ///     <term>Yes</term>
+        ///     <term> Contents of the ledger entry. </term>
+        ///   </item>
+        ///   <item>
+        ///     <term>subLedgerId</term>
+        ///     <term>string</term>
+        ///     <term></term>
+        ///     <term> Identifier for sub-ledgers. </term>
+        ///   </item>
+        ///   <item>
+        ///     <term>transactionId</term>
+        ///     <term>string</term>
+        ///     <term></term>
+        ///     <term> A unique identifier for the state of the ledger. If returned as part of a LedgerEntry, it indicates the state from which the entry was read. </term>
+        ///   </item>
+        /// </list>
+        /// </remarks>
+        /// <param name="content"> The content to send as the body of the request. </param>
+        /// <param name="subLedgerId"> The sub-ledger id. </param>
+        /// <param name="waitForCompletion"> If <c>true</c>, the <see cref="PostLedgerEntryOperation"/> will not be returned until the ledger entry is committed.
+        /// If <c>false</c>,<see cref="Operation.WaitForCompletionResponse(System.Threading.CancellationToken)"/> must be called to ensure the operation has completed.</param>
```

Comment:
should the `UpdateStatus` method be added here too and change the docstring to something like... "If false, use blah or blah to poll the results to ensure the operation completes"